### PR TITLE
feat(Attribute): support generic value for items

### PR DESCRIPTION
**Pull request by BotellaA**

### CODE REVIEW

File: include/geode/basic/attribute_utils.h [hunk 1]
```diff
@@ -195,4 +217,40 @@ namespace geode
     IMPLICIT_GENERIC_ATTRIBUTE_CONVERSION( unsigned int );
     IMPLICIT_GENERIC_ATTRIBUTE_CONVERSION( float );
     IMPLICIT_GENERIC_ATTRIBUTE_CONVERSION( double );
+
+#define IMPLICIT_ARRAY_GENERIC_ATTRIBUTE_CONVERSION( Type )                    \
+    template < size_t dimension >                                              \
```
Comment (panquez) [inline include/geode/basic/attribute_utils.h:222]: Is dimension a good name? I mean: it is maybe confusing with 2- or 3- dimension used for coordinates.

**Review by panquez — APPROVED** [review]

File: include/geode/basic/attribute_utils.h (hunk 1 above)
Comment (MelchiorSchuh) [reply include/geode/basic/attribute_utils.h:222]: I agree, maybe nb_items or another more explicit name ?

File: include/geode/basic/attribute_utils.h [hunk 2]
```diff
@@ -195,4 +217,40 @@ namespace geode
...
+                "incorrect item value" );                                      \
+            return static_cast< float >( value[item] );                        \
+        }                                                                      \
+                                                                               \
+        static bool is_genericable()                                           \
+        {                                                                      \
+            return true;                                                       \
+        }                                                                      \
+        static local_index_t nb_items()                                        \
+                                                                               \
```
Comment (MelchiorSchuh) [inline include/geode/basic/attribute_utils.h:245]: remove line break

File: tests/basic/test-attribute.cpp [hunk 3]
```diff
@@ -376,10 +387,16 @@ void test_generic_value( geode::AttributeManager& manager )
             std::array< double, 3 > >(
             "array_double", std::array< double, 3 >() );
     array_attr->set_value( 2, { 3.1, 1.3 } );
-    OPENGEODE_EXCEPTION( !array_attr->is_genericable(),
+    OPENGEODE_EXCEPTION( array_attr->is_genericable(),
         "[Test] Foo attribute is not genericable" );
```
Comment (MelchiorSchuh) [inline tests/basic/test-attribute.cpp:391]:
```suggestion
        "[Test] Foo attribute is genericable" );
```

File: tests/basic/test-attribute.cpp [hunk 4]
```diff
@@ -376,10 +387,16 @@ void test_generic_value( geode::AttributeManager& manager )
             std::array< double, 3 > >(
             "array_double", std::array< double, 3 >() );
     array_attr->set_value( 2, { 3.1, 1.3 } );
```
Comment (MelchiorSchuh) [inline tests/basic/test-attribute.cpp:389]: Is it not strange to initialize an array of 3 doubles with only 2 values ?

**Review by MelchiorSchuh — APPROVED** [review]

**Pull request merged**